### ISSUE TITLE
HasFunction method evaluates regular knots as True

### ISSUE THREAD

**Issue by WolvenBard** [inkle/ink-unity-integration#209]
### Context:

I have two ink containers, one a regular knot, and the other a function:

```
=== goblin_talk ===
...

=== function has_revived ===
...
```
If I call upon `Story.cs`'s method `HasFunction`, it evaluates both `goblin_talk` and `has_revived` as functions.

### Problem:

I'm looking at the HasFunction defined in Story.cs. It looks like its not exclusive to just functions, and actually returns True for values that represent regular knots (not functions).

Looking at the implementation of the method, this makes sense as it is calling upon KnotContainerWithName which in turn is just searching the contents of the main container:
public Runtime.Container KnotContainerWithName (string name)
{
    INamedContent namedContainer;
    if (mainContentContainer.namedContent.TryGetValue (name, out namedContainer))
        return namedContainer as Container;
    else
        return null;
}

Is this an oversight? 

### Attempted Solutions:

Looking at the containers themselves, there doesn't really seem to be a clear way to discern if a container is a regular Knot or a Function. The only thing I can see is that if I search within the contents of the container, Functions tend to have a `ControlCommand` entry labeled `PopFunction`

This may be one way to discern, but seems hacky. Not sure if that `PopFunction` could appear elsewhere other than an explicit function knot.

**Comment by tomkail** [comment]
Hi! Can you please post this on the main Ink repo? This one is mostly just
a Unity wrapper for it. I agree that this wants improving though, it’s
gotten me in the past too. Ta!

**Comment by WolvenBard** [comment]
@tomkail  Certainly, happy to repost this issue on there as well.

Would you like me to replicate this pull request on the main repo as well?
https://github.com/inkle/ink-unity-integration/pull/210

I assume the fix would want to be on both repos?

**Comment by tomkail** [comment]
Yes please! I’ll merge it in if Joe accepts the commit on the ink repo. Ta!

**Comment by WolvenBard** [comment]
Thanks @tomkail. I've reposted here:

**Issue:** https://github.com/inkle/ink/issues/928
**Pull Request:** https://github.com/inkle/ink/pull/927

**Comment by Necronomison** [comment]
Hi @tomkail, I see you marked this one as complete.

Though I still see this MR open on the main repo:
https://github.com/inkle/ink/pull/927

**Comment by tomkail** [comment]
Sorry I just meant to close it, not mark it complete. This will be pulled in once it's approved on the main ink repo!
